### PR TITLE
Upgrade to Relay v2.0.0 #85

### PR DESCRIPTION
**Pull request by artola**
- update dependencies accordingly Relay v2.0.0
  - removed unused dependency @types/immutable
  - added minimalist types for RelayCompilerPublic
  - updated fixtures
  - updated test TypeScriptGenerator to support directives as v2.0.0

### CODE REVIEW

File: types/relay-compiler/lib/RelayCompilerPublic.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import * as GraphQLCompilerTypes from "graphql-compiler";
```
Comment (sibelius) [inline types/relay-compiler/lib/RelayCompilerPublic.d.ts]: graphql-compiler was merged in relay-compiler

should we remove graphql-compiler at all?

Comment (artola) [reply types/relay-compiler/lib/RelayCompilerPublic.d.ts]: we have not a dependency `graphql-compiler` ... it is transitive from `relay-compiler`. But we need the typings `graphql-compiler`, then I just re-exported them as `RelayCompilerPublic`.

**Comment by sibelius** [comment]
not sure how to update your branch

I've tested here https://github.com/entria/entria-fullstack, and it is working fine

I think we should automate releases based on this https://github.com/intuit/auto

**Comment by artola** [comment]
@sibelius Great!

@sibelius @alloy Could we proceed with #86 releasing manually a v2.0 package? As there is a breaking change that is related to relay-compiler v1 changing to v2 (and other dependencies version related), it makes sense to go for v2 too.
People that is wanting to upgrade Relay depends on this package for easy integration.

**Comment by alloy** [comment]
Agreed that we should have automatic releases, but for now I'll manually cut a release 👍

**Comment by artola** [comment]
@alloy I agree with automatic releases too, just that it should not delay a new release as people is willing to move into Relay v2 asap.
Thanks for your help.

**Comment by alloy** [comment]
Done, thanks for all the help! 👌

**Pull request merged**